Readme.md

```diff
@@ -1,12 +1,528 @@
+At Criteo we use chef, but none of the followings relies on the use of a specific IAAS solution.
+
+### Backend services sizing
+
+The load produced by Garmadon heavily depends on your cluster size, occupancy, the nature of containers you run, and Garmadon's own properties.
+
+To help you dimension your ElasticSearch and Kafka cluster here are some stats that you should consider.
+
+* Every NodeManager of your cluster will send 
+  * 2 CONTAINER_MONITORING_EVENT per container actually running on the machine every time containers are monitored. This frequency is directly bound to the parameter _yarn.nodemanager.container-monitor.interval-ms_ defined in *yarn-site.xml*
+  * 17 OS events every 30 seconds (configurable)
+* Every container will send
+  * 1 JVMSTATS_EVENT every 10 seconds (configurable)
+  * 1 APPLICATION_STATE_EVENT every 10 second per application running on the RM
+  * FS_EVENT when it performs HDFS actions
+  * GC_EVENT when gc occurs
+* Every spark container will in addition send Spark events from the Spark LiveListenerBus
+
+The size observed for those events on our production cluster:
+
+| EventType   | Size observed in bytes (quantile 0.99) | Comment |
+| -----------:| --------------------------------------:| -------:|
+| CONTAINER_MONITORING_EVENT | 261 | predictive value |
+| JVMSTATS_EVENT / OS_EVENT / GC_EVENT | 1920 | predictive value |
+| SPARK_TASK_EVENT | 157786 | depends partly on the actual running application |
+| FS_EVENT | 800 | mostly depends on paths sizes |
+| STATE_EVENT | missing value | predictive value |
+
+Some of the load is predictible (and tunable) but a big part of the load is also very depending on the number of container you run, their life span, the framework you use, etc...
+
+We provide you what we observe on our production cluster. It is a 2900 nodes cluster on which we run around 25 million containers per day.
+
+On this cluster, Garmadon generates :
+* around 45k evts/sec
+* from 9TB to 10TB of data per day on Kafka (not including replication)
+* from 30GB to 40GB per hour in ElasticSearch (not including replication)
+
+Here is the event repartition on ElasticSearch cluster over a 4h period
+
+![](doc/images/es/event_type_repartition.png)
+
+### Install Garmadon Reader for Elastic Search
+
+#### Configure Elastic Search
+
+You can use our template [readers/elasticsearch/src/main/elasticsearch/es/template.json](readers/elasticsearch/src/main/elasticsearch/es/template.json) as the index template and configure *number_of_shards* and *number_of_replica*
+to fit your needs.
+
+The current code expects the use of a _hourly index_.
+
+#### Configure Grafana
+
+Your Elastic Search cluster has to be configured as a datasource for Grafana.
+
+Then you can import JSON from [readers/elasticsearch/src/main/elasticsearch/grafana](readers/elasticsearch/src/main/elasticsearch/grafana) in your Grafana and use the Elastic Search datasource.
```

Comment:
Current dashboards expects that datasource is named garmadon<anychars>.
You will then see it on the datasource variable of each dashboards (we used it to choose different cluster)